### PR TITLE
add mutmut

### PR DESCRIPTION
**Pull request by carlodri**
https://github.com/boxed/mutmut

https://mutmut.readthedocs.io/en/latest/

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Review by carlodri — COMMENTED** [review]
🤦‍♂️  sorry for the silly mistake.... and thanks for your correction @mbargull!

**Comment by carlodri** [comment]
For me this is ready to merge.

File: recipes/mutmut/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  entry_points:
+    - mutmut = mutmut.__main__:climain
+
+# until https://github.com/conda-forge/pony-feedstock/pull/20 we need to constrain python <3.10
+requirements:
+  host:
+    - python >=3.7, <3.10
+    - pip
+  run:
+    - python >=3.7, <3.10
```
Comment (dopplershift) [inline recipes/mutmut/meta.yaml:25]: With the referenced PR merged you shouldn't need the `<3.10` constraint, no?

Comment (carlodri) [reply recipes/mutmut/meta.yaml:25]: yes, you are perfectly right!

Comment (carlodri) [reply recipes/mutmut/meta.yaml]: looks like the CDN has not been updated yet to use the new build [`noarch/pony-0.7.14-pyhd8ed1ab_1.tar.bz2`](https://anaconda.org/conda-forge/pony/files)

**Comment by dopplershift** [comment]
Ok, because it's trying to build on 3.10, it still grabs the old package because that "satisfies" it's requirements. I've submitted [a patch](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/184) to the metadata that should fix that. Once that's merged and propagated, we can see what happens here.

**Comment by carlodri** [comment]
oh you're right. Thanks a lot for the PR!

**Comment by carlodri** [comment]
@conda-forge-admin restart CI

**Review by dopplershift — APPROVED** [review]

**Pull request merged**